src/views/OrderDetail.vue

```diff
@@ -0,0 +1,187 @@
+                <p class="ion-text-wrap">{{ $t("SKU") }}: {{ item.sku }}</p>
+              </ion-label>
+            </ion-item>
+            <ion-item>
+              <ion-label v-for="property in item.properties" :key="property">
+                <p>{{ property.name }}: {{ property.value }}</p>
+              </ion-label>
+            </ion-item>
+            <ion-radio-group :value="isSelected(item)" @ionChange="addProperty(item, $event)">
+              <ion-item lines="none" class="border-top">
+                <ion-label>{{ $t("Pre Order") }}</ion-label>
+                <ion-radio value="Pre Order" />
+              </ion-item>
+              <ion-item lines="none" class="border-top">
+                <ion-label>{{ $t("Back Order") }}</ion-label>
+                <ion-radio value="Back Order" />
+              </ion-item>
+              <ion-item lines="none" class="border-top">
+                <ion-label>{{ $t("None") }}</ion-label>
+                <ion-radio value="None" />
+              </ion-item>
+            </ion-radio-group>
+            
+          </ion-card>
+        </main>
+        <ion-button @click="updateOrder(order.line_items)">{{ $t("Update") }}</ion-button>
+      </div>
+    </ion-content>
+  </ion-page>
+</template>
+<script lang="ts">
+import {
+  IonBadge,
+  IonButton,
+  IonCard,
+  IonContent,
+  IonHeader,
+  IonIcon,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonPage,
+  IonTitle,
+  IonToolbar,
+  IonBackButton,
+  IonRadio,
+  IonRadioGroup
+} from "@ionic/vue";
+import {
+  sendOutline,
+  swapVerticalOutline,
+  callOutline,
+  mailOutline,
+} from "ionicons/icons";
+import { defineComponent } from 'vue';
+import { mapGetters, useStore } from 'vuex';
+import { useRouter } from 'vue-router';
+import { DateTime } from 'luxon';
+
+export default defineComponent({
+  name: 'Home',
+  components: {
+    IonBadge,
+    IonButton,
+    IonCard,
+    IonContent,
+    IonHeader,
+    IonIcon,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonPage,
+    IonTitle,
+    IonToolbar,
+    IonBackButton,
+    IonRadio,
+    IonRadioGroup
+  },
+  computed: {
+    ...mapGetters({
+      order: 'order/getDraftOrder'
+    })
+  },
+  created(){
+    this.store.dispatch('order/fetchDraftOrder', this.$route.query.id);
+  },
+  methods: {
+    addProperty (item: any, event: any) {
+      if(item.properties?.find((property: any) => property.name === 'Note')){
+        const index = item.properties?.findIndex((property: any) => property.name === 'Note');
+        item.properties.splice(index, 1);
+      }
+      if(!(event.detail.value === "None")){
+        item.properties.push({ name: 'Note', value: event.detail.value })
+      }
+    },
+    updateOrder (lineItems: any) {
+      const id = this.$route.query.id;
+      this.store.dispatch('order/updateDraftOrder', {lineItems, id});
+    },
+    isSelected (item: any) {
+      if (item.properties?.find((property: any) => property.name === 'Note')?.value === "Pre Order"){
```

Comment:
Store the logic to find the note property in a variable and then use that variable to check for `Pre Order` or `Back Order`.

Reply:
Also, we can directly return the value of the property and can remove the check to compare its value, and in case of no property found we can return None.

Reply:
Updated the condition